hibernate-core/src/main/java/org/hibernate/event/jfr/internal/JfrEventManager.java

```diff
@@ -0,0 +1,390 @@
+/*
+ * Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later.
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.event.jfr.internal;
+
+import java.sql.PreparedStatement;
+import java.util.UUID;
+
+import org.hibernate.cache.spi.Region;
+import org.hibernate.cache.spi.access.CachedDomainDataAccess;
+import org.hibernate.engine.spi.SharedSessionContractImplementor;
+import org.hibernate.event.jfr.CacheGetEvent;
+import org.hibernate.event.jfr.CachePutEvent;
+import org.hibernate.event.jfr.JdbcBatchExecutionEvent;
+import org.hibernate.event.jfr.JdbcConnectionAcquisitionEvent;
+import org.hibernate.event.jfr.JdbcConnectionReleaseEvent;
+import org.hibernate.event.jfr.JdbcPreparedStatementCreationEvent;
+import org.hibernate.event.jfr.JdbcPreparedStatementExecutionEvent;
+import org.hibernate.event.jfr.SessionClosedEvent;
+import org.hibernate.event.jfr.SessionOpenEvent;
+import org.hibernate.internal.build.AllowNonPortable;
+import org.hibernate.persister.collection.CollectionPersister;
+import org.hibernate.persister.entity.EntityPersister;
+import org.hibernate.stat.internal.StatsHelper;
+
+import static java.util.concurrent.TimeUnit.NANOSECONDS;
+
+@AllowNonPortable
+public class JfrEventManager {
+
+	public static SessionOpenEvent beginSessionOpenEvent() {
+		final SessionOpenEvent sessionOpenEvent = new SessionOpenEvent();
+		if ( sessionOpenEvent.isEnabled() ) {
+			sessionOpenEvent.begin();
+		}
+		return sessionOpenEvent;
+	}
+
+	public static void commitSessionOpenEvent(
```

Comment:
Would the name `#completeXYZ` be better to correlate better with `#beginXYZ`?  

JFR calls the end operation commit, so I understand that naming too.

Reply:
I agree `completeXYZ` is better